### DISCUSSION TITLE
Error TypeError: ExtPay is not a function

### DISCUSSION THREAD

**Discussion by PaePongsiriTri** [Glench/ExtPay#70] [Q&A] [answered]
I use Typescript and Webpack bundle
I follow instructions until 3. Add ExtPay to background.js and then I got an error 

background.ts
`const ExtPay = require("ExtPay")`
`const extpay = ExtPay("sample-extension")`

 TypeError: ExtPay is not a function
Any Idea?

This is my webpack code

webpack.common.js
const path = require('path');
const CopyPlugin = require('copy-webpack-plugin');
const HtmlPlugin = require('html-webpack-plugin');
const { CleanWebpackPlugin } = require('clean-webpack-plugin');

module.exports = {
  entry: {
    popup: path.resolve('src/popup/popup.tsx'),
    options: path.resolve('src/options/options.tsx'),
    background: path.resolve('src/background/background.ts'),
    contentScript: path.resolve('src/contentScript/contentScript.ts'),
    context: path.resolve('src/contentScript/context.ts'),
  },
  module: {
    rules: [
      {
        test: /\.tsx?$/,
        use: 'ts-loader',
        exclude: /node_modules/,
      },
      {
        test: /\.css$/i,
        use: ['style-loader', 'css-loader'],
      },
      {
        test: /\.(jpg|jpeg|png|woff|woff2|eot|ttf|svg)$/,
        type: 'asset/resource'
      }
    ]
  },
  resolve: {
    extensions: ['.tsx', '.ts', '.js'],
  },
  plugins: [
    new CleanWebpackPlugin({
      cleanStaleWebpackAssets: false,
    }),
    new CopyPlugin({
      patterns: [
        {
          from: path.resolve('src/static'),
          to: path.resolve('dist'),
        }
      ]
    }),
    ...getHtmlPlugins([
      'popup',
      'options'
    ]),
  ],
  output: {
    filename: '[name].js',
    path: path.resolve('dist'),
  },
  optimization: {
    splitChunks: {
      chunks(chunk) {
        return chunk.name !== 'contentScript' && chunk.name !== 'background' && chunk.name != 'context'
      }
    },
  }
}

function getHtmlPlugins(chunks) {
  return chunks.map(chunk => new HtmlPlugin({
    title: 'React Extension',
    filename: `${chunk}.html`,
    chunks: [chunk],
  }))
}

Webpack.dev.js

const { merge } = require('webpack-merge');
const common = require('./webpack.common.js');

module.exports = merge(common, {
  mode: 'development',
  devtool: 'cheap-module-source-map',
})

**Comment by Glench** [comment, marked as answer]
It sounds like an issue with your webpack bundling. Maybe post your config and see if others can help since I don't know much about webpack

**Reply by PaePongsiriTri** [reply]
> It sounds like an issue with your webpack bundling. Maybe post your config and see if others can help since I don't know much about webpack

Sure! I added to the question
